src/backend/distributed/executor/local_executor.c

```diff
@@ -512,7 +664,22 @@ TaskAccessesLocalNode(Task *task)
 
 
 /*
- * ErrorIfTransactionAccessedPlacementsLocally() errors out if a local query on any shard
+ * ErrorIfTransactionAndAnyTaskAccessedPlacementsLocally errors out if a local query
+ * on any shard has already been executed in the same transaction and any of the tasks
+ * in the taskList will access a local placement.
+ */
+void
+ErrorIfTransactionAndAnyTaskAccessedPlacementsLocally(List *taskList)
```

Comment:
Two notes on this function:
- The name is not easy to understand, can you come-up with something more intuitive?
- Can we call this in other places instead of `ErrorIfTransactionAccessedPlacementsLocally `? I think not easily for copy as there are tasks. Still, can you consider the 3 other places that we use  `ErrorIfTransactionAccessedPlacementsLocally `?

Reply:
> Can we call this in other places instead of ErrorIfTransactionAccessedPlacementsLocally ? I think not easily for copy as there are tasks. Still, can you consider the 3 other places that we use ErrorIfTransactionAccessedPlacementsLocally ?

As we have no tasks generated on those pieces of code, I guess it wouldn't be that automatic to using the function I introduced in those places as well

Reply:
> TaskTrackerExecScan

Task-tracker is not enabled by default, so I'm fine with not improving that. But, we should be able to get taskList easily there 


> analyze

Don't we have tasks for analyze?

Reply:
We call `ErrorIfTransactionAccessedPlacementsLocally` for analyze at the very beginning of utility hook. In this case, I should make some small changes on utility hook. Working on it

Reply:
I think we can punt on it if this is going to create trouble for now, we can look later, as a follow-up PR

Reply:
removed that block to test the behavior in that case, 3min

Reply:
I guess it wouldn't be that straightforward to execute analyze locally (bunch of distributed deadlocks here https://circleci.com/gh/citusdata/citus/94822), skipping it ?

Reply:
or should I work on making it work ?

Reply:
let's skip for now, I think